src/storage/in_mem_storage_structure/in_mem_column_chunk.cpp

```diff
@@ -48,6 +48,10 @@ uint32_t InMemColumnChunk::getDataTypeSizeInColumn(common::LogicalType& dataType
     }
 }
 
+void InMemColumnChunk::copyChunk(std::shared_ptr<arrow::RecordBatch> batch) {
```

Comment:
Add an assertion here batch only contains one column. Also, you can ignore passing `nullptr` here, otherwise, add a inline comment here `copyArrowArray(*batch->column(0), nullptr /* nodeOffsets */);`.

How about call this function `copyArrowBatch` instead of `copyChunk`? I think the former one is more accurate here.